polus-precompute-volume-plugin/src/main.py

```diff
@@ -18,7 +18,32 @@
 logger = logging.getLogger("main")
 logger.setLevel(logging.INFO)
 
-def main():
+def main(input_dir: str,
+         output_dir: str,
+         imagetype: str,
+         imagepattern: str,
+         mesh: bool):
+    
+    # Get list of images that we are going to through
+    # Get list of output paths for every image
+    logger.info("\n Getting the images...")
+    fp_images = fp(Path(input_dir),imagepattern)
+    input_images = []
+    output_images = []
+    for i in fp_images():
+        image = i[0]['file']
+        if os.path.exists(image):
+            input_images.append(image)
+            output_images.append(os.path.join(output_dir, os.path.basename(image)))
+    assert len(input_images) == len(output_images)
+
+    # Build one pyramid for each image in the input directory
+    with ProcessPoolExecutor() as executor:
+        executor.map(utils.build_pyramid, input_images, output_images, repeat(imagetype), repeat(mesh))
```

Comment:
Without specifying `max_workers`, it defaults to the number of CPUs. If you are going to do this, I would recommend limiting the number of workers for the ThreadPoolExecutor to 1 or 2 since you may overload the processor or disk, resulting in worse performance.